src/pages/Create.js

```diff
@@ -0,0 +1,424 @@
+                aria-haspopup={true}
+                onMouseEnter={handleOpenPop}
+                onMouseLeave={handleClosePop}
+            >
+                <a href={props.leaf.linkUrl} onClick={handleClick} style={{cursor:'pointer', color:'black'}}>
+                    {props.children}
+                </a>
+            </Typography>
+            <Popper open={popoverAnchor!=null}
+                     anchorEl={popoverAnchor}
+            >
+                <p>{props.leaf.linkUrl}</p>
+            </Popper>
+        </>
+    );
+}
+
+const Leaf = props => {
+    return (
+        <span
+            {...props.attributes}
+            style={{ fontWeight: props.leaf.bold ? 'bold' : 'normal', fontStyle: props.leaf.italic ? 'italic' : 'normal',
+                    textDecorationLine: props.leaf.underlined ? 'underline' : null}}
+        >
+      {props.leaf.link ? (
+          <LinkElement {...props} />
+          ) : props.children}
+    </span>
+    )
+}
+
+const toggleFormat = (editor, format, linkUrl = null) => {
+    const isActive = isFormatActive(editor, format);
+    if (linkUrl) {
+        Transforms.setNodes(
+            editor,
+            { [format]: isActive ? null : true, 'linkUrl': isActive ? null : linkUrl},
+            { match: Text.isText, split: true }
+        );
+    } else {
+        Transforms.setNodes(
+            editor,
+            { [format]: isActive ? null : true},
+            { match: Text.isText, split: true }
+        );
+    }
+
+}
+
+const formatTitle = (event, editor) => {
+    event.preventDefault();
+    const formatSelection = titleFormatSelector(editor);
+    // Toggle the block type depending on whether there's already a match.
+    Transforms.setNodes(
+        editor,
+        { type: formatSelection==='paragraph' ? 'paragraph' : formatSelection==='title' ? 'title' : 'subtitle'},
+        { match: n => Editor.isBlock(editor, n) }
+    );
+}
+
+const titleFormatSelector = (editor) => {
+    let [match] = Editor.nodes(editor, {
+        match: n => n.type === 'title',
+    });
+    if (!!match) {
+        return 'subtitle';
+    } else {
+        [match] = Editor.nodes(editor, {
+            match: n => n.type === 'subtitle'
+        });
+
+        if (!!match) {
+            return 'paragraph';
+        }
+    }
+
+    return 'title';
+}
+
+const formatBlock = (editor, formatType) => {
+    const isActive = isTypeActive(editor, formatType);
+    Transforms.setNodes(editor,
+        {type: isActive ? 'paragraph' : formatType},
+        {match: n => Editor.isBlock(editor, n)}
+    );
+
+}
+
+const isTypeActive = (editor, formatType) => {
+    const [match] = Editor.nodes(editor, {
+        match: n => n.type === formatType
+    });
+
+    return !!match;
+}
+
+const isFormatActive = (editor, format) => {
+    const [match] = Editor.nodes(editor, {
+        match: n => n[format] === true,
+        mode: 'all'
+    });
+
+    return !!match;
+}
+
+const HoveringToolbar = () => {
```

Comment:
thinking forward, the user will probably want to be able to edit the blog. So the functionality of editing the blog should probably be refactored.

`HoveringToolbar` then could probably be refactored into its own component.